WebAPI/Controllers/ParcelController.cs

```diff
@@ -25,21 +26,23 @@ public async Task<ActionResult<IEnumerable<Parcel>>> Get()
             return Ok(result);
         }
 
-        [HttpGet("parcel/{id}")]
-        public async Task<ActionResult<Parcel>> GetParcel(int id)
+        [HttpGet("parcel/{trackingNumber}")]
+        public async Task<ActionResult<Parcel>> GetParcel(string trackingNumber)
         {
-            _logger.LogInformation("Executed {0}->{1}({2})", this.GetType().Name, ControllerContext.ActionDescriptor.ActionName, id); ; //testing purposes
-            var res = await _service.GetAllByUserId(id);
+            _logger.LogInformation("Executed {0}->{1}({2})", this.GetType().Name, ControllerContext.ActionDescriptor.ActionName, trackingNumber); ; //testing purposes
+            var res = await _service.GetByTrackingId(trackingNumber);
 
             return Ok(res);
         }
 
         [HttpPost("parcel")]
         public async Task<ActionResult<Parcel>> Post(Parcel parcel)
         {
-            await _service.Insert(parcel);
-            _logger.LogInformation("Executed {0}->{1}", (this.GetType().Name, ControllerContext.ActionDescriptor.ActionName) +parcel.ToString()); //testing purposes
-            return Ok();
+            TrackingIdGenerator generator = new();
```

Comment:
Dependency injection should be used for Generator, because creating with constructor is not the "good practice" way

Reply:
Inject in `Program.cs`, receive in this class' constructor